### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: loggestd/src/usage_monitor.rs [hunk 1]
```diff
@@ -39,6 +41,47 @@ impl From<Statvfs> for SpaceData {
...
+#[cfg(unix)]
+fn get_fs_data(directory: &Path) -> Result<SpaceData, io::Error> {
+    Ok(statvfs(directory)
+        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
+        .into())
+}
+
+#[cfg(windows)]
+fn get_fs_data(directory: &Path) -> Result<SpaceData, io::Error> {
+    use widestring::U16CString;
```
Comment (drrlvn) [inline loggestd/src/usage_monitor.rs]: Why use this over `OsString`?

File: loggestd/src/args.rs [hunk 2]
```diff
@@ -10,11 +12,17 @@ pub struct Opt {
...
         long,
         parse(from_os_str),
         default_value = "/run/loggestd.sock",
         env = "LOGGESTD_SOCKET"
     )]
     pub unix_socket: PathBuf,
+
+    /// Address to listen to
+    #[cfg(windows)]
+    #[structopt(long, default_value = "127.0.0.1:1337", env = "LOGGESTD_LISTEN")]
```
Comment (drrlvn) [inline loggestd/src/args.rs]: Use 1099 as default port.

File: loggestd/src/usage_monitor.rs [hunk 3]
```diff
@@ -39,6 +41,47 @@ impl From<Statvfs> for SpaceData {
...
+    use winapi::shared::minwindef::FALSE;
+    use winapi::um::{errhandlingapi::GetLastError, fileapi, winnt::ULARGE_INTEGER};
+
+    let wstr = U16CString::from_os_str(directory.as_os_str()).unwrap();
+
+    let mut avail: ULARGE_INTEGER = Default::default();
+    let mut total: ULARGE_INTEGER = Default::default();
+    let mut free: ULARGE_INTEGER = Default::default();
+
+    if FALSE
```
Comment (drrlvn) [inline loggestd/src/usage_monitor.rs]: Reverse condition arguments.

File: loggestd/src/usage_monitor.rs [hunk 4]
```diff
@@ -39,6 +41,47 @@ impl From<Statvfs> for SpaceData {
...
+        == unsafe {
+            fileapi::GetDiskFreeSpaceExW(
+                wstr.as_ptr(),
+                &mut avail as *mut _,
+                &mut total as *mut _,
+                &mut free as *mut _,
```
Comment (drrlvn) [inline loggestd/src/usage_monitor.rs]: Can you pass `NULL` here?

File: loggestd/src/usage_monitor.rs (hunk 1 above)
Comment (r-darwish) [reply loggestd/src/usage_monitor.rs]: https://retep998.github.io/blog/winapi-0.3/

This page (and another blog post I saw) suggest collecting the OsStr into a vector, pushing the null termination and then get that vector as bytes. I think that I rather have a crate deal this for us

**Pull request merged**